switch kmod-6.12-nvidia-r570 to kernel-6.12-devel

bcressey opened the pull request:
**Issue number:**
Fixes fallout from the mid-air collision between #95 and #99. 

**Description of changes:**
Switch the 570 kmod to use files from `kernel-6.12-devel` instead of the now-absent `kernel-6.12-archive`.


**Testing done:**
Built locally. Also tested a `g5g.8xlarge` launch and confirmed that the Tesla kmod could be linked and loaded.


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

Commit:
kmod-6.12-nvidia-r570: switch to kernel-6.12-devel

The 6.12 kernel's "devel" subpackage has the required kernel sources
already unpacked, so we can dispense with the archive extraction.

The "devel" subpackage also provides symlinks for the major kernel
version, so we don't need to discover the release field at build
time.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
kernel-6.12: unrestrict System.map for devel package

It's not useful to restrict the permissions on a file used at build
time, and it interferes with the ability of downstream packages to
copy the kernel development files to a writable location.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

yeazelm commented on packages/kmod-6.12-nvidia-r570/kmod-6.12-nvidia-r570.spec:
```diff
@@ -179,9 +182,8 @@ install -d %{buildroot}%{_cross_factorydir}%{_cross_sysconfdir}/{drivers,ld.so.c
-KERNEL_VERSION=$(cat %{kernel_sources}/include/config/kernel.release)
 sed \
-  -e "s|__KERNEL_VERSION__|${KERNEL_VERSION}|" \
+  -e "s|__KERNEL_VERSION__|%{kernel_major}|" \
```
I don't believe the path we are swapping out for this has the symlink we would need here, we need the M.m.p for the kernel, not just M.m.
```
ls /x86_64-bottlerocket-linux-gnu/sys-root/usr/lib/modules
6.12.20
```
From `nvidia-tmpfiles.conf.in`:
```
nvidia-tmpfiles.conf.in
1:R __PREFIX__/lib/modules/__KERNEL_VERSION__/kernel/drivers/extra/video/nvidia/tesla - - - - -
2:d __PREFIX__/lib/modules/__KERNEL_VERSION__/kernel/drivers/extra/video/nvidia/tesla 0755 root root - -
3:R __PREFIX__/lib/modules/__KERNEL_VERSION__/kernel/drivers/extra/video/nvidia/open-gpu - - - - -
4:d __PREFIX__/lib/modules/__KERNEL_VERSION__/kernel/drivers/extra/video/nvidia/open-gpu 0755 root root - -
```

arnaldo2792 replied:
@bcressey added a change in #99 to install a symlink using just the major version.

yeazelm replied:
Oh nice! I missed this change in the other PR.

arnaldo2792 commented:
Builds are passing, we can merge and close #117 , right @rpkelly ?

bcressey commented:
This is failing at runtime with:
```
[    7.085272] driverdog[2408]: 16:52:35 [ERROR] '/usr/bin/ld' failed - stderr: /usr/bin/ld: cannot open linker script file /usr/src/kernels/6.12.20/scripts/module.lds: No such file or directory
[FAILED] Failed to start Link Tesla kernel modules.
```

So I need to fix up the kernel devel package.

rpkelly commented:
> Builds are passing, we can merge and close #117 , right @rpkelly ?

If this merges we will not need #117

Commit:
kernel-6.12: add module linker script to base package

driverdog depends on this script to link modules at runtime.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
kernel-6.12: adjust permissions for devel files

Allow the builder user to modify development files and directories,
so that downstream kmod builds can rebuild host tools without making
an extra copy of the sources.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

bcressey commented:
⬆️ force push removes the need for `rsync` to make a second copy of the kernel development files, and adds the `module.lds` linker script that `driverdog` depends on.

yeazelm reviewed (approved):
Nice!
